connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -219,7 +230,7 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
                                     log.trace("Wrote record successfully: topic {} partition {} offset {}",
                                             recordMetadata.topic(), recordMetadata.partition(),
                                             recordMetadata.offset());
-                                    commitTaskRecord(record);
+                                    commitTaskRecord(preTransformRecord);
```

Comment:
It shouldn't be a 2x increase since we are not closing over the transformed `record` in this callback.